Shared/Samples/Style symbols from mobile style file/StyleSymbolsFromMobileStyleFileView.SymbolOptionsListView.swift

```diff
@@ -0,0 +1,121 @@
+        
+        /// The view model for the sample.
+        @ObservedObject var model: Model
+        
+        /// The list of all the symbols and their associated data from the symbol style.
+        @State private var symbols: [SymbolDetails] = []
+        
+        var body: some View {
+            VStack {
+                Image(uiImage: model.currentSymbol?.image ?? UIImage())
+                
+                List {
+                    // Create a picker for each symbol category.
+                    ForEach(SymbolCategory.allCases, id: \.self) { category in
+                        Section(category.label) {
+                            Picker("Symbol Names", selection: $model.symbolOptionSelections.categoryKeys[category]) {
+                                ForEach(symbols.filter { $0.category == category.rawValue }, id: \.key) { symbol in
+                                    Label {
+                                        Text(symbol.label)
+                                    } icon: {
+                                        Image(uiImage: symbol.image)
+                                    }
+                                    .tag(symbol.key as String?)
+                                }
+                            }
+                            .pickerStyle(.inline)
+                            .labelsHidden()
+                        }
+                    }
+                    
+                    Section {
+                        ColorPicker("Color", selection: $model.symbolOptionSelections.color)
+                        VStack {
+                            HStack {
+                                Text("Size")
+                                Spacer()
+                                Text(model.symbolOptionSelections.size.formatted(.number.rounded()))
+                                    .foregroundColor(.secondary)
+                            }
+                            Slider(value: $model.symbolOptionSelections.size, in: 20...60, step: 1)
+                        }
+                    }
+                }
+            }
+            .task(id: model.symbolOptionSelections) {
+                // Update the current symbol when an option selection changes.
+                await model.updateCurrentSymbol(displayScale: displayScale)
+            }
+            .task(id: displayScale) {
+                // Update the symbols when the display scale changes.
+                symbols = await getSymbols(displayScale: displayScale)
+                await model.updateCurrentSymbol(displayScale: displayScale)
+            }
+        }
+        
+        /// Gets the symbols from the symbol style.
+        /// - Parameter displayScale: The display scale of the environment for creating symbol swatches.
+        /// - Returns:  An `Array` of `SymbolDetails` which contain a symbol and its associated information.
+        private func getSymbols(displayScale: Double) async -> [SymbolDetails] {
```

Comment:
I would consider moving this function to the model. Also storing the symbols on the model could simplify some things. You have already a model, then have state for the symbols. Why not just put the symbols on the model

Reply:
I didn't put the symbols on the model since they are only accessed by one view, which is also why I have that function as part of the view and not the model. I was under the impression that it made it more encapsulated for lack of a better word. It may easy to understand/cleaner if it was part of the model, though. Let me know if you think I should move everything to model.

Reply:
@rolson did you still think it would be better to move these things to the model?

Reply:
Since the error handling was added, it does make sense to move more to the model so that code that calls these things doesn't have to do/catch and set an error on the model.

For the error handling, the methods don't have to be throwing per se, but could just set an error on the model.

Reply:
I moved everything to the model and added an `updateDisplayScale(displayScale:)` method that prevents everything from reloading when the display scale doesn't actually change. I also removed the error checking from `updateCurrentSymbol()` since it would throw a cancellation error if you changed the symbols too quickly. Thanks for all your feedback!